connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -118,7 +119,11 @@ private[sql] class AvroOptions(
    * taken into account. If the former one is not set too, the `snappy` codec is used by default.
    */
   val compression: String = {
-    parameters.get(COMPRESSION).getOrElse(SQLConf.get.avroCompressionCodec)
+    val v = parameters.get(COMPRESSION).getOrElse(SQLConf.get.avroCompressionCodec)
+    if (v == null) {
```

Comment:
This problem is actually here and there, not only `AvroOptions`. I believe a lot of `CSVOptions` also has this problem.

Reply:
+1 for @HyukjinKwon 's comment.

Reply:
Would there be compatibility issues if null values are prohibited in `CaseInsensitiveMap`?